master/docs/developer/utils.rst

```diff
@@ -886,6 +886,44 @@ For example, a particular daily scheduler could be configured on multiple master
         Therefore, in this method it is safe to reassign the "active" status to another instance.
         This method may return a Deferred.
 
+.. py:class:: SingletonService
+
+    This class implements a generic Service that needs to be instantiated as a singleton according to its parameters.
```

Comment:
if you can have n instances of this service but with different parameters this is not a singleton... for accessing to external services, we use normal services "mounted" into a known proxy (with an hardcoded name, which is not very elegant), for instance to be able to have 1 gerrit service talking to Gerrit 2.6 and another talking to Gerrit 2.11.